prepare_GBM_data.R

```diff
@@ -0,0 +1,195 @@
+                        help = "Processed sequencing data output file path"),
+  optparse::make_option("--clinical_input",
+                        default = NULL,
+                        help = "Clinical information input file path (Excel file)"),
+  optparse::make_option("--clinical_output",
+                        default = NULL,
+                        help = "Clinical information output file path (.tsv)"),
+  optparse::make_option("--overwrite",
+                        action = "store_true",
+                        default = FALSE,
+                        help = "Overwrite existing output files [default: %default]")
+)
+
+opt <- optparse::parse_args(optparse::OptionParser(option_list=option_list))
+source("util/option_functions.R")
+check_options(opt, "prepare_GBM_data.R") # how to get this script name easily?
+
+# load libraries
+suppressMessages(library(tidyverse))
+
+# set options
+tcga_seq_expression_input_filepath <- opt$seq_input
+gbm_array_expression_input_filepath <- opt$array_input
+metadata_json_input_filepath <- opt$metadata_input
+gbm_array_output_filepath <- opt$array_output
+gbm_seq_output_filepath <- opt$seq_output
+clinical_xlxs_input_filepath <- opt$clinical_input
+clinical_xlxs_output_filepath <- opt$clinical_output
+
+################################################################################
+# Array data
+################################################################################
+
+# read in refine.bio GBM array expression data
+gbm_array_expression <- read_tsv(gbm_array_expression_input_filepath,
+                                 col_types = cols(
+                                   .default = col_double(),
+                                   Gene = col_character()
+                                 ))
+
+# load up aggregated metadata json file
+metadata_json <- jsonlite::fromJSON(metadata_json_input_filepath,
+                                    simplifyVector = FALSE)
+
+# accession IDs present in expression data
+available_array_accession_ids <- colnames(gbm_array_expression)[-1]
+
+# starting with flattened metadata, parse out raw TCGA IDs and filter for tumors
+all_array_tumor_samples <- tibble(accession = names(metadata_json$samples)) %>%
+  filter(accession %in% available_array_accession_ids) %>% # check these are ==
+  rowwise() %>%
+  mutate(tcga_id_raw = metadata_json$samples[[accession]]$refinebio_annotations[[1]]$characteristics_ch1[[9]] %>%
+           str_remove("sample: ")) %>%
+  mutate(tcga_id = str_sub(tcga_id_raw, 1, 12),
+         sample = str_sub(tcga_id_raw, 14, 15)) %>%
+  filter(str_starts(sample, "0")) %>% # remove non-tumor samples
+  ungroup()
+
+# keep one (first) accession per TCGA ID
+array_accession_tcga_id_keep <- all_array_tumor_samples %>%
+  group_by(tcga_id) %>%
+  summarize(accession = sort(accession)[1])
+accession_colnames_keep <- colnames(gbm_array_expression)[-1][colnames(gbm_array_expression)[-1] %in% array_accession_tcga_id_keep$accession]
```

Comment:
If this array data is coming from refine.bio, I don't _know_ if we can expect for the accessions to be in the same order every time. I'm not sure what's to be done about that (track IDs that _do get used_ and then check against that downstream?), but seems worth noting somewhere.

Reply:
Ah, but the `sort()` step should take care of this, which I missed the first time through. (I came back to suggest something like this specifically!)